appsec/src/extension/request_lifecycle.c

```diff
@@ -404,7 +404,7 @@ static void _reset_globals()
         _superglob_equiv = NULL;
     }
 
-    if (_client_ip && _client_ip != CLIENT_IP_LOOKUP_FAILED) {
+    if (_client_ip) {
```

Comment:
Shouldn't the fix instead be something like:

```diff
diff --git a/appsec/src/extension/request_lifecycle.c b/appsec/src/extension/request_lifecycle.c
index 3964f5c2b..50ed798c3 100644
--- a/appsec/src/extension/request_lifecycle.c
+++ b/appsec/src/extension/request_lifecycle.c
@@ -413,8 +413,8 @@ static void _reset_globals()
 
     if (_client_ip && _client_ip != CLIENT_IP_LOOKUP_FAILED) {
         zend_string_release(_client_ip);
-        _client_ip = NULL;
     }
+    _client_ip = NULL;
 
     if (Z_TYPE(_blocking_function) != IS_UNDEF) {
         zval_ptr_dtor(&_blocking_function);
```

o/wise we keep trying (and faling) to extract the ip